File: packages/stealth-sdk/src/constants/schemes.ts
import type { SchemeId } from '../types'

/**
 * EIP-5564 Stealth Address Scheme IDs
 *
 * Scheme ID 0 is reserved.
 * Scheme ID 1 is secp256k1 with view tags.
 */
export const SCHEME_ID = {
  /** Reserved scheme */
  RESERVED: 0 as SchemeId,
  /** secp256k1 with view tags (recommended) */
  SECP256K1: 1 as SchemeId,
} as const

/**
 * Default scheme ID for StableNet
 */
export const DEFAULT_SCHEME_ID: SchemeId = SCHEME_ID.SECP256K1

/**
 * View tag size in bytes
 */
export const VIEW_TAG_SIZE = 1

/**
 * Ephemeral public key size (compressed) in bytes
 */
export const COMPRESSED_PUBKEY_SIZE = 33

/**
 * Ephemeral public key size (uncompressed) in bytes
 */
export const UNCOMPRESSED_PUBKEY_SIZE = 65

/**
 * Stealth meta address prefix
 */
export const STEALTH_META_ADDRESS_PREFIX = 'st'

/**
 * Supported chain prefixes
 */
export const CHAIN_PREFIX = {
  ETHEREUM: 'eth',
  STABLENET: 'stablenet',
  SEPOLIA: 'sep',
  BASE_SEPOLIA: 'basesep',
} as const

export type ChainPrefix = (typeof CHAIN_PREFIX)[keyof typeof CHAIN_PREFIX]

File: packages/sdk-go/plugins/stealth/constants/schemes.go
// Package constants provides constant values for the stealth address plugin.
package constants

// ViewTagSize is the size of the view tag in bytes.
const ViewTagSize = 1

// CompressedPubKeySize is the size of a compressed public key in bytes.
const CompressedPubKeySize = 33

// UncompressedPubKeySize is the size of an uncompressed public key in bytes.
const UncompressedPubKeySize = 65

// StealthMetaAddressSize is the size of a stealth meta address in bytes.
// (spending public key + viewing public key = 33 + 33 = 66 bytes)
const StealthMetaAddressSize = 66

// StealthMetaAddressPrefix is the URI prefix for stealth meta addresses.
const StealthMetaAddressPrefix = "st"

// MaxMetadataSize is the maximum size of metadata in bytes.
const MaxMetadataSize = 1024

// ChainPrefix constants for common chains.
const (
	ChainPrefixEthereum    = "eth"
	ChainPrefixStableNet   = "stablenet"
	ChainPrefixSepolia     = "sep"
	ChainPrefixBaseSepolia = "basesep"
)
